addons/laserpointer/XEH_postInit.sqf

```diff
@@ -7,32 +7,54 @@
 if (!hasInterface) exitWith {};
 
 GVAR(nearUnits) = [];
+GVAR(index) = -1;
+GVAR(laserClassesCache) = [] call CBA_fnc_createNamespace;
+GVAR(redLaserUnits) = [];
+GVAR(greenLaserUnits) = [];
 
 #include "initKeybinds.sqf"
 
 ["ace_settingsInitialized", {
-    //If not enabled, dont't add draw eventhandler or PFEH (for performance)
-    if (!GVAR(enabled)) exitWith {};
+    // if not enabled, dont't add draw eventhandler or PFEH (for performance)
+    if !(GVAR(enabled)) exitWith {};
 
-    // @todo. Maybe move to common?
     [{
-        // handle RHS / bugged vehicle slots
-        private _camPosAGL = positionCameraToWorld [0,0,0];
-        if !((_camPosAGL select 0) isEqualType 0) exitWith {};
-
-        private _nearUnits = [];
-        {
-            _nearUnits append crew _x;
-            if (count _nearUnits > 10) exitWith {
-                _nearUnits resize 10;
+        GVAR(nearUnits) = call FUNC(getNearUnits);
+    }, 5, []] call CBA_fnc_addPerFrameHandler;
+
+    // custom scheduler
+    [{
+        GVAR(index) = GVAR(index) + 1;
+        private _unit = GVAR(nearUnits) param [GVAR(index), objNull];
+
+        if (isNull _unit) exitWith {
+            GVAR(index) = -1;
+            GVAR(redLaserUnits) = [];
+            GVAR(greenLaserUnits) = [];
+        };
+
+        private _weapon = currentWeapon _unit;
+        private _laser = (_unit weaponAccessories _weapon) select 1;
+
+        if (_laser != "") then {
+            private _laserID = GVAR(laserClassesCache) getVariable _laser;
+
+            if (isNil "_laserID") then {
+                _laserID = getNumber (configFile >> "CfgWeapons" >> _laser >> "ACE_laserpointer");
+                GVAR(laserClassesCache) setVariable [_laser, _laserID];
             };
-        } forEach nearestObjects [_camPosAGL, ["AllVehicles"], 50]; // when moving this, search also for units inside vehicles. currently breaks the laser in FFV
 
-        GVAR(nearUnits) = _nearUnits;
+            if (_unit isFlashlightOn _weapon) then {
+                if (_laserID isEqualTo 1) exitWith {
+                    GVAR(redLaserUnits) pushBackUnique _unit;
+                };
 
-    } , 5, []] call CBA_fnc_addPerFrameHandler;
+                if (_laserID isEqualTo 2) exitWith {
+                    GVAR(greenLaserUnits) pushBackUnique _unit;
+                };
+            };
+        };
+    }, 0.1, []] call CBA_fnc_addPerFrameHandler;
 
-    addMissionEventHandler ["Draw3D", {
-        call FUNC(onDraw);
-    }];
+    addMissionEventHandler ["Draw3D", FUNC(onDraw)];
```

Comment:
`GVAR(onDrawEH) =`
Like i wanted to do right now because of the performance Problem. The missionmaker knows if any player can have a laserpointer right now. If no one has one he can disable the EH and readd it once its possible for players to have a laserpointer.